### ISSUE TITLE
Weird DP behaviour on AWS P2 16xlarge with Wandb Logger

### ISSUE THREAD

**Issue by aced125** [PyTorchLightning/pytorch-lightning#2251]
### Before asking:   
1. search the issues.   
2. search the docs.    

#### What is your question?
Wandb Logger seems to not log all steps.

AWS P2 16xLarge has 16 K80s on 1 node.

I am using dp distrib mode.

#### Code
My training steps:
```
    def training_step(self, batch, batch_idx):
        loss, logits, *_ = self(batch)
        output = {"loss": loss, "log": {"loss": loss}}
        return output
```

My logs:
![wandb-logs](https://user-images.githubusercontent.com/44452903/85087100-6db54200-b1d4-11ea-9cb8-da5e47de5e6d.png)

When I change my `training_step` to explicitly log metrics, all works again:
```
    def training_step(self, batch, batch_idx):
        loss, logits, *_ = self(batch)
        self.logger.log_metrics({"loss": loss.cpu()})
        output = {"loss": loss}
        return output
```
![working-wandb-logs](https://user-images.githubusercontent.com/44452903/85087574-c507e200-b1d5-11ea-98fb-f58b5b92f09f.png)

#### What's your environment?

 - OS: [e.g. iOS, Linux, Win]: Linux
 - Packaging [e.g. pip, conda]: Pip
 - Version [e.g. 0.5.2.1]: 0.8.0

**Comment by Laksh1997** [comment]
I can confirm I don't think this is the issue of multi-gpu or anything.

Seems to be something with the wandb logger.

Below is a pic from training on CPU. Global step is 800 while x axis step is 25. Logging seems to be happening sporadically.

<img width="1449" alt="Screenshot 2020-06-19 at 19 57 50" src="https://user-images.githubusercontent.com/59830552/85171815-a4886800-b267-11ea-9684-0598daa65b9b.png">

**Comment by Borda** [comment]
@borisdayma mind have a look? 🐰

**Comment by borisdayma** [comment]
@aced125 Do you have an example that I can reproduce?

**Comment by borisdayma** [comment]
I tried [a test](https://colab.research.google.com/drive/16d1uctGaw2y9KhGBlINNTsWpmlXdJwRW?usp=sharing) and everything seems to work fine.

`step` increases at each logging on W&B for pytorch-lightning integration (due to #2015).
You just need to set the x-axis as `global_step`.

![image](https://user-images.githubusercontent.com/715491/96060550-6c301980-0e56-11eb-9354-5019c74176a5.png)
